frame/tx-pause/src/mock.rs

```diff
@@ -0,0 +1,236 @@
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = pallet_balances::AccountData<u64>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = ();
+	type OnSetCode = ();
+	type MaxConsumers = frame_support::traits::ConstU32<16>;
+}
+
+parameter_types! {
+	pub const ExistentialDeposit: u64 = 1;
+	pub const MaxLocks: u32 = 10;
+}
+impl pallet_balances::Config for Test {
+	type Balance = u64;
+	type DustRemoval = ();
+	type RuntimeEvent = RuntimeEvent;
+	type ExistentialDeposit = ExistentialDeposit;
+	type AccountStore = System;
+	type WeightInfo = ();
+	type MaxLocks = MaxLocks;
+	type MaxReserves = ();
+	type ReserveIdentifier = [u8; 8];
+	type FreezeIdentifier = ();
+	type RuntimeHoldReason = RuntimeHoldReason;
+	type MaxHolds = ConstU32<0>;
+	type MaxFreezes = ConstU32<0>;
+}
+
+impl pallet_utility::Config for Test {
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeCall = RuntimeCall;
+	type PalletsOrigin = OriginCaller;
+	type WeightInfo = ();
+}
+
+/// Mocked proxies to check that the tx-pause also works with the proxy pallet.
+#[derive(
+	Copy,
+	Clone,
+	Eq,
+	PartialEq,
+	Ord,
+	PartialOrd,
+	Encode,
+	Decode,
+	RuntimeDebug,
+	MaxEncodedLen,
+	scale_info::TypeInfo,
+)]
+pub enum ProxyType {
+	Any,
+	JustTransfer,
+	JustUtility,
+}
+
+impl Default for ProxyType {
+	fn default() -> Self {
+		Self::Any
+	}
+}
+
+impl InstanceFilter<RuntimeCall> for ProxyType {
+	fn filter(&self, c: &RuntimeCall) -> bool {
+		match self {
+			ProxyType::Any => true,
+			ProxyType::JustTransfer => {
+				matches!(c, RuntimeCall::Balances(pallet_balances::Call::transfer { .. }))
+			},
+			ProxyType::JustUtility => matches!(c, RuntimeCall::Utility { .. }),
+		}
+	}
+	fn is_superset(&self, o: &Self) -> bool {
+		self == &ProxyType::Any || self == o
+	}
+}
+
+impl pallet_proxy::Config for Test {
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeCall = RuntimeCall;
+	type Currency = Balances;
+	type ProxyType = ProxyType;
+	type ProxyDepositBase = ConstU64<1>;
+	type ProxyDepositFactor = ConstU64<1>;
+	type MaxProxies = ConstU32<4>;
+	type WeightInfo = ();
+	type CallHasher = BlakeTwo256;
+	type MaxPending = ConstU32<2>;
+	type AnnouncementDepositBase = ConstU64<1>;
+	type AnnouncementDepositFactor = ConstU64<1>;
+}
+
+parameter_types! {
+	pub const MaxNameLen: u32 = 50;
+}
+
+frame_support::ord_parameter_types! {
+	pub const PauseOrigin: u64 = 1;
+	pub const UnpauseOrigin: u64 = 2;
+}
+
+/// Calls that are never allowed to be paused.
+#[derive(Copy, Clone, Encode, Decode, RuntimeDebug, MaxEncodedLen, scale_info::TypeInfo)]
+pub struct WhitelistedCalls;
+impl Contains<RuntimeCallNameOf<Test>> for WhitelistedCalls {
+	fn contains(full_name: &RuntimeCallNameOf<Test>) -> bool {
+		let unpausables: Vec<RuntimeCallNameOf<Test>> = vec![(
+			b"Balances".to_vec().try_into().unwrap(),
+			b"transfer_keep_alive".to_vec().try_into().unwrap(),
+		)];
+
+		unpausables.contains(full_name)
```

Comment:
Is this possible to write as a `BTreeSet`? If so, would that be a better data structure to use as an example here? (assuming people may copy this)

Reply:
Ah could be. Dan and me changed the scope and data layout multiple times, so maybe it is possible again...

Reply:
Using match now instead of constructing a vector:  
```rust
match (full_name.0.as_slice(), full_name.1.as_slice()) {
	(b"Balances", b"transfer_keep_alive") => true,
	_ => false,
}
```
My earlier comment was about the `PausedCalls` storage item.